src/support/generation.c

```diff
@@ -159,15 +162,30 @@ __wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
              */
             if (minutes == 0) {
                 minutes = 1;
-                __wt_seconds(session, &start);
+                __wt_epoch(session, &start);
             } else {
-                __wt_seconds(session, &stop);
-                if (stop - start > minutes * WT_MINUTE) {
+                __wt_epoch(session, &stop);
+                time_diff = WT_TIMEDIFF_MS(stop, start);
+                if (time_diff > minutes * WT_MINUTE * WT_THOUSAND) {
                     __wt_verbose_notice(session, WT_VERB_GENERATION,
                       "%s generation drain waited %u minutes", __gen_name(which), minutes);
                     ++minutes;
                     WT_ASSERT(session, minutes < 4);
                 }
+                /* Enable extra logs before timing out. */
```

Comment:
Same here.

Reply:
Added in a09ffd452cb4ecc33171291465fb2d8f992b60fc